### PR TITLE
Prototype form-builder routing

### PR DESCRIPTION
**Pull request by bryan-robitaille**
# Summary | Résumé

Enables routing through the URL to a specific action and form using the pattern `/form-builder/${action}/{formID}`
This PR also sets the initial values of the Navigation and Template stores through props which will reduce the number of page renders required before interactivity is ready for the user.

# Unresolved questions / Out of scope | Questions non résolues ou hors sujet

Some existing forms will throw errors when loaded into form-builder due to existing complex components or missing description fields.  Any additional work to make the feature more robust will be in a future PR.

### CODE REVIEW

File: components/form-builder/layout/Publish.tsx [hunk 1]
```diff
@@ -34,6 +37,7 @@ export const Publish = () => {
     if (result && result?.error) {
       setError(true);
       return;
+      s;
```
Comment (timarney) [inline components/form-builder/layout/Publish.tsx]: typo?

Comment (bryan-robitaille) [reply components/form-builder/layout/Publish.tsx]: Just pushed a commit to fix.  A stray keyboard command gone awry.

**Comment by timarney** [comment]
Looks good @bryan-robitaille 

Per you note we'll end up with some errors for description fields  etc.. but this moves things forward nicely

Are you looking for an approval to merge at this point?  Thinking that would be best so we can get some other WIP PRs updated to use this.

**Comment by bryan-robitaille** [comment]
Let's merge so we can start using the new Store pattern and I'll create a new PR to fix the issues importing some of the forms into Form Builder.

**Review by timarney — APPROVED** [review]
🥇

**Comment by timarney** [comment]
FYI - I started experimenting with debounce on https://github.com/cds-snc/platform-forms-client/pull/1194 

With this PR starting the move to selectors + some further updates we can stop all the extra renders for key "panel" updates.

**Pull request merged**